signer/hsm.go

```diff
@@ -0,0 +1,67 @@
+package signer
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"fmt"
+	"io"
+
+	"github.com/miekg/pkcs11"
+	"github.com/mozilla-services/autograph/crypto11"
+)
+
+type HSM interface {
+	GetPrivateKey(label []byte) (crypto.PrivateKey, error)
+	MakeKey(keyTpl interface{}, keyName string) (crypto.PrivateKey, crypto.PublicKey, error)
+	GetRand() io.Reader
+}
+
+type GenericHSM struct {
```

Comment:
I will note that there's an argument to be made that this shouldn't exist, and the trivial implementations of `GetPrivateKey` and `GetRand` should just stay in `signer.go`. My opinion is that if we're striving to _fully_ isolate (ie: move all `crypto11` interactions) here that we should keep this. If we don't care about that, we can drop this and simply have `AWSHSM` (and later `GCPHSM`) implement `MakeKey`.

Reply:
Oh, in another part of the code, my feedback included wanting to use this interface type as an argument! I think it's worthy.